decidim-core/app/presenters/decidim/log/base_presenter.rb

```diff
@@ -0,0 +1,150 @@
+            action_string,
+            i18n_params
+          ).html_safe +
+            " (#{h.localize(action_log.created_at, format: :decidim_short)})"
+        end
+      end
+
+      private
+
+      attr_reader :action_log, :view_helpers
+      alias h view_helpers
+
+      delegate :action, :user, :resource, :feature, :participatory_space, to: :action_log
+
+      # Private: Presents a space. If the space is found in the database, it
+      # links to it. Otherwise it only shows the name.
+      #
+      # Returns an HTML-safe String.
+      def present_space
+        return present_space_name if participatory_space.blank?
+
+        h.link_to(present_space_name, space_path)
+      end
+
+      # Private: Presents the resource of the action. If the space is found
+      # in the database, it links to it. Otherwise it only shows the name.
+      #
+      # Returns an HTML-safe String.
+      def present_resource
+        return present_resource_name if resource.blank?
+
+        h.link_to(present_resource_name, resource_path)
+      end
+
+      # Private: Finds the link for the given space.
+      #
+      # Returns an HTML-safe String.
+      def space_path
+        Decidim::ResourceLocatorPresenter.new(participatory_space).path
+      end
+
+      # Private: Finds the link for the given resource.
+      #
+      # Returns an HTML-safe String.
+      def resource_path
+        Decidim::ResourceLocatorPresenter.new(resource).path
+      end
+
+      # Private: Presents a space. If the space is found in the database, it
+      # links to it. Otherwise it only shows the name.
+      #
+      # Returns an HTML-safe String.
+      def present_space_name
+        h.translated_attribute action_log.extra["participatory_space"]["title"]
+      end
+
+      # Private: Presents a space. If the space is found in the database, it
+      # links to it. Otherwise it only shows the name.
+      #
+      # Returns an HTML-safe String.
+      def present_resource_name
+        h.translated_attribute action_log.extra["resource"]["title"]
+      end
+
+      # Private: Presents a space. If the space is found in the database, it
+      # links to it. Otherwise it only shows the name.
+      #
+      # Returns an HTML-safe String.
+      def present_user
+        return present_user_name if user.blank?
+        h.link_to(present_user_name, h.decidim.profile_path(action_log.extra["user"]["nickname"]))
+      end
+
+      # Private: Presents a space. If the space is found in the database, it
+      # links to it. Otherwise it only shows the name.
+      #
+      # Returns an HTML-safe String.
+      def present_user_name
+        name = action_log.extra["user"]["name"]
+        nickname = action_log.extra["user"]["nickname"]
+        "#{name} @#{nickname}"
+      end
+
+      # Private: Finds the name of the I18n key that will be sued for the
+      # current log action.
+      #
+      # Returns a String.
+      def action_string
+        case action
```

Comment:
Add `.to_s` to prevent possible errors when comparing against `Nil` or if someone returns a symbol? (yay types!)